Docs: Revise IAM policies with more information (#989)
redpanda-data/documentation: versioned_docs/version-22.2/platform/security/iam-roles.mdx

1. [Create an S3 storage bucket](https://docs.aws.amazon.com/AmazonS3/latest/userguide/creating-bucket.html).
2. [Create an IAM policy](https://aws.amazon.com/blogs/security/writing-iam-policies-how-to-grant-access-to-an-amazon-s3-bucket/).
3. [Create an IAM role and assign the policy to that role](https://docs.aws.amazon.com/IAM/latest/UserGuide/access_policies_job-functions_create-policies.html).
4. Tiered Storage with AWS requires that the user have the following permissions to read and create objects on the bucket to be used with the cluster (or on all buckets): `GetObject`, `DeleteObject`, `PutObject`, `PutObjectTagging`, `ListBucket`. 
5. [Bind the VM](https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/iam-roles-for-amazon-ec2.html#attach-iam-role), or [Pod in the case of Kubernetes](https://docs.aws.amazon.com/eks/latest/userguide/specify-service-account-role.html), to the IAM role.

:::caution

> `GetObject`, `DeleteObject`, `PutObject`, `PutObjectTagging`, and `ListBucket` are the least actions required to fully utilize the cloud storage as of the v22.3 release, but you may need to update it for future redpanda releases. If the bucket is dedicated to redpanda, we recommend allowing all (*) actions within the bucket.

Reviewer: ```suggestion
Starting in v22.3, `GetObject`, `DeleteObject`, `PutObject`, `PutObjectTagging`, and `ListBucket` are the least actions required to fully utilize cloud storage. If the bucket is dedicated to Redpanda, we recommend allowing all (*) actions within the bucket.
```

Author: Addressed in b8bd431

> :::caution
> `GetObject`, `DeleteObject`, `PutObject`, `PutObjectTagging`, and `ListBucket` are the least actions required to fully utilize the cloud storage as of the v22.3 release, but you may need to update it for future redpanda releases. If the bucket is dedicated to redpanda, we recommend allowing all (*) actions within the bucket.
> :::

Reviewer: ```suggestion
:::caution
`GetObject`, `DeleteObject`, `PutObject`, `PutObjectTagging`, and `ListBucket`  are not required to fully utilize the cloud storage, but may be required in future Redpanda releases. Hence, if the bucket is dedicated to Redpanda, we recommend allowing all (*) actions within the bucket.
:::
```

Author: Addressed in b8bd431

#### Sample full access IAM policy

The following example policy grants full access to the `test` S3 bucket. 

```
{
    "Version": "2012-10-17",
    "Statement": [
        {
            "Effect": "Allow",
            "Action": [
                "s3:*"
            ],
            "Resource": [
	        "arn:aws:s3:::test",
	        "arn:aws:s3:::test/*"
	    ]
        }
    ]
}
```

#### Sample minimum read/write IAM policy

> This policy represents a minimum IAM policy for the `test` S3 buckets required when using the Tiered Storage functionalities. 

Reviewer: ```suggestion
This policy represents a minimum IAM policy for the `test` S3 buckets required when using Tiered Storage. 
```

Author: Addressed in b8bd431

```
{
    "Version": "2012-10-17",
    "Statement": [
        {
            "Effect": "Allow",
            "Action": [
                "s3:PutObject",
                "s3:PutObjectTagging",
                "s3:GetObject",
                "s3:DeleteObject"
            ],
            "Resource": "arn:aws:s3:::test/*"
        },
        {
            "Effect": "Allow",
            "Action": [
                "s3:ListBucket"
            ],
            "Resource": "arn:aws:s3:::test"
        }
    ]
}
```

#### Sample read-only IAM policy

> A more restrictive "read-only" IAM policy is shown below. This policy only allows a user to get and list objects in the `test` S3 bucket. Such a policy could be used for a read replica topic on a remote cluster that hosts read replica topics, but not Tiered Storage topics.

Reviewer: not sure why read-only is in quotation marks?
```suggestion
A more restrictive read-only IAM policy is shown below. This policy only allows a user to get and list objects in the `test` S3 bucket. Such a policy could be used for a read replica topic on a remote cluster that hosts read replica topics, but not Tiered Storage topics.
```

Author: Addressed in b8bd431

2. [Create an IAM policy](https://aws.amazon.com/blogs/security/writing-iam-policies-how-to-grant-access-to-an-amazon-s3-bucket/).
3. [Create an IAM role and assign the policy to that role](https://docs.aws.amazon.com/IAM/latest/UserGuide/access_policies_job-functions_create-policies.html).
4. Tiered Storage with AWS requires that the user have the following permissions to read and create objects on the bucket to be used with the cluster (or on all buckets): `GetObject`, `DeleteObject`, `PutObject`, `PutObjectTagging`, `ListBucket`. 
5. [Bind the VM](https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/iam-roles-for-amazon-ec2.html#attach-iam-role), or [Pod in the case of Kubernetes](https://docs.aws.amazon.com/eks/latest/userguide/specify-service-account-role.html), to the IAM role.

#### Sample full access IAM policy

The following example policy grants full access to the `test` S3 bucket. 

```
{
    "Version": "2012-10-17",
    "Statement": [
        {
            "Effect": "Allow",
            "Action": [
                "s3:*"
            ],
            "Resource": [
	        "arn:aws:s3:::test",
	        "arn:aws:s3:::test/*"
	    ]
        }
    ]
}
```

> #### Sample minimum (least) IAM policy

Reviewer 2: Can we just link back to the other page rather than duplicating this?

Author: This is a 22.2 specific page, not 22.3.

Reviewer 2: Ah, I see.
